druid-shell/src/mouse.rs

```diff
@@ -48,22 +48,49 @@ pub enum MouseButton {
     X1,
     /// Second X button.
     X2,
+    /// Unknown button.
+    Unknown,
 }
 
-impl MouseButton {
+/// Adds convenience methods to `MouseButton` and `Option<MouseButton>`.
+pub trait MouseButtonExt {
```

Comment:
I get the motivation but this doesn't feel great to me. I do think I might prefer `MouseButton::None`, and we could sort of redefine `MouseButton` to be 'the mouse button that triggered a given event, if one exists.' really this is all about mouse moved. Down and up must be associated with a specific button, but moved can have no or all buttons, etc. I wonder if it's worth representing differently in the API?

I've also experimented with ways to build a [higher-level API on top of this](https://github.com/linebender/runebender/blob/master/src/mouse.rs) in runebender; I'm not totally happy with the results, but I think something like that _could_ be useful?